pinot-broker/src/test/java/org/apache/pinot/broker/requesthandler/SelectStarWithOtherColsRewriteTest.java

```diff
@@ -0,0 +1,90 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.broker.requesthandler;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.request.Expression;
+import org.apache.pinot.common.request.PinotQuery;
+import org.apache.pinot.sql.parsers.CalciteSqlParser;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class SelectStarWithOtherColsRewriteTest {
+
+  private static final Map<String, String> COL_MAP;
+
+  static {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    builder.put("homeRuns", "homeRuns");
+    builder.put("playerStint", "playerStint");
+    builder.put("groundedIntoDoublePlays", "groundedIntoDoublePlays");
+    builder.put("playerID", "playerID");
+    builder.put("$segmentName", "$segmentName");
+    builder.put("$docId", "$docId");
+    builder.put("$hostName", "$hostName");
+    COL_MAP = builder.build();
+  }
+
+  @Test
+  public void testHappyCase() {
+    String sql = "SELECT $segmentName,*,$hostName FROM baseballStats";
+    PinotQuery pinotQuery = CalciteSqlParser.compileToPinotQuery(sql);
+    BaseBrokerRequestHandler.updateColumnNames("baseballStats", pinotQuery, false, COL_MAP);
+    List<Expression> newSelections = pinotQuery.getSelectList();
+    Assert.assertEquals(newSelections.get(0).getIdentifier().getName(), "$segmentName");
+    Assert.assertEquals(newSelections.get(1).getIdentifier().getName(), "homeRuns");
+    Assert.assertEquals(newSelections.get(2).getIdentifier().getName(), "playerStint");
+    Assert.assertEquals(newSelections.get(3).getIdentifier().getName(), "groundedIntoDoublePlays");
+    Assert.assertEquals(newSelections.get(4).getIdentifier().getName(), "playerID");
+    Assert.assertEquals(newSelections.get(5).getIdentifier().getName(), "$hostName");
+  }
+
+  @Test
+  public void testDupCols() {
+    String sql = "SELECT playerID,homeRuns,* FROM baseballStats";
+    PinotQuery pinotQuery = CalciteSqlParser.compileToPinotQuery(sql);
+    BaseBrokerRequestHandler.updateColumnNames("baseballStats", pinotQuery, false, COL_MAP);
+    List<Expression> newSelections = pinotQuery.getSelectList();
```

Comment:
This is not correct / compliant with Standard SQL as far as I can tell. The column list in the select expressions should be deduplicated. So, `playerID` and `homeRuns` should be returned only once. 

Even generally for non star queries, using the same column name twice in the SELECT clause should be deduplicated. 

However, the scenario of aliasing must be handled.

For example, the query `SELECT CustomerID AS C1, * FROM Customers` must return CustomerID column in the result twice (with result column name as C1 and CustomerID respectively).

Reply:
@siddharthteotia thanks for the review. Actually I wasn't aware that this is dictated by the SQL standard. Had asked both of these questions (deduplication and returning internal columns) in the original [issue](https://github.com/apache/pinot/issues/7867#issuecomment-1001020152) for clarification. Will address these.

Reply:
Tried this with Postgres, same behaviour (column is returned twice). The ANSI SQL standard doesn't allow for extra columns to be added but DBs have their own extensions to it. I think we should go with how MySQL and Postgres are doing it, that is, returning the columns twice unless there are some implications.

@siddharthteotia

Reply:
In the above query `mysql> select *, cases, abs(cases) from covid_cases`, cases and abs(cases) are different since the latter is using function. By deduplication, I mean to do only for the identifiers (simple column names) in the SELECT list. 

My preference would be to deduplicate and not return the column twice as the output is cleaner that way. So, in the above query cases column should be returned exactly once. 

If the user wants to get the same column twice, they should use aliasing (like I mentioned earlier)

`SELECT CustomerID AS C1, * FROM Customers`

CustomerID should be returned twice -- once as C1 and next as CustomerID

But if we want to stick to how MySQL or Postgres is doing and that is the general ANSI SQL / user expectation, then let's not deduplicate. I am ok with it. Just does not look clean

Trying the query here seems to deduplicate -- https://www.w3schools.com/sql/trysql.asp?filename=trysql_select_all

It says it uses MySQL but not sure which database it is using underneath

Reply:
@amrishlal @suddendust